strs.go

```diff
@@ -69,6 +69,31 @@ func (db *RoseDB) SetEx(key, value []byte, duration time.Duration) error {
 	return err
 }
 
+// SetNX sets the key-value pair if it is not exist. It returns nil if the key
+// exists in db.
+func (db *RoseDB) SetNX(key, value []byte) error {
+	db.strIndex.mu.Lock()
+	defer db.strIndex.mu.Unlock()
+
+	val, err := db.getVal(key)
+
```

Comment:
check this err:
`if err != nil && err != ErrKeyNotFound {return err}`

Reply:
I did in ac86516. I used `errors.Is()` instead of `==`. [Here's](https://github.com/golang/go/wiki/ErrorValueFAQ#:~:text=If%20you%20currently%20compare%20errors%20using%20%3D%3D%2C%20use%20errors.Is%20instead.) why.